app/src/main/java/com/everest/userinfo/validations/ValidateFields.kt

```diff
@@ -0,0 +1,58 @@
+package com.everest.userinfo.validations
+
+
+import android.content.Context
+import android.util.Patterns
+import android.widget.Toast
+import com.everest.userinfo.HomeActivity
+
+class ValidateFields constructor(private var context: Context) {
+
+    fun inputs(email:String, phNo:String, pinCode:String): Boolean {
+        if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
```

Comment:
We can make a very small improvement here that we can separate showing toast and validating field separate.

Since we are calling ValidateFields in viewmodel, we can make it not depend on context. you can post an toast/error in viewmodel and in view you can observe it to show toast